remixer_core/src/main/java/com/google/android/libraries/remixer/Remixer.java

```diff
@@ -16,13 +16,25 @@
 
 package com.google.android.libraries.remixer;
 
+import com.google.android.libraries.remixer.sync.SynchronizationMechanism;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Locale;
 
 /**
- * Contains a list of {@link Variable}es.
+ * Contains a list of {@link Variable}s or {@link Trigger}s.
+ *
+ * <p>The Remixer object is the heart and soul of Remixer as a framework, it coordinates syncing and
+ * value changes.
+ *
+ * <p>For value and data syncing/persistence (both locally across activities, and persisting/saving
+ * to cloud backends, etc) and keeping a global state, the Remixer object relies on a {@link
+ * SynchronizationMechanism} set at initialization time (usually in your Application.onCreate()).
+ *
+ * <p>If you do not set a SynchronizationMechanism, remixer will not try to sync values across
+ * different contexts (activities) or check for consistency among those. It is recommended you
+ * always use one (even if it is just the LocalValueSyncing one) so you catch consistency errors.
```

Comment:
Ugh, no, can't unless we move the whole remixer_storage into remixer_core, because LocalValueSyncing is in remixer_storage which depends on remixer_core, can't have a circular dependency.

We have two options:

# Move the current contents of remixer_storage into remixer_core
-Makes it possible to have a default
-Makes it possible to keep some of those private methods package-local
-Kinda pollutes remixer_core with serialization logic.
-Remixer_storage/sync will still be necessary because firebase cloud sdk has to be used from an android_library, so some sync logic would be in remixer_core and some in remixer_storage/sync.

# Keep it as is
-storage and sync is all centralized in a separate module, does not pollute remixer_core
-impossible to have a default.
-need to keep some of those methods public (triggerWithoutNotifiyingOthersWithTheSameKey, setValueWithoutNotifyingOthersWithTheSameKey) :/

# Conclusion

I kinda prefer moving this sync logic to remixer_core, even if it feels kinda out of place when the things that depend on android SDK are put in a separate project.

What do you think?